.github/workflows/smoke-tests.yml

```diff
@@ -0,0 +1,233 @@
+        required: true
+        type: string
+      target_url:
+        description: Base URL for the application
+        required: true
+        type: string
+      user_email:
+        description: Test user email
+        required: true
+        type: string
+      user_password:
+        description: Test user password
+        required: true
+        type: string
+      course_id:
+        description: Course ID to test viewing the course page
+        required: true
+        type: string
+      target_plugins:
+        description: Desired plugins to validate
+        required: true
+        type: string
+      unit_id:
+        description: Unit ID to test complete-multiplechoice-unit
+        required: true
+        type: string
+
+
+jobs:
+  check-heartbeat:
+    if: ${{ inputs.run_test == 'All' ||  inputs.run_test == 'check-heartbeat'}}
+    name: Check Heartbeat
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout actions-hub repo
+        uses: actions/checkout@v4
+        with:
+          repository: nelc/actions-hub
+          ref: and/add-basic-smoke-tests
+          path: actions-hub
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+
+      - name: Install dependencies
+        run: |
+          pip install pytest-playwright
+          playwright install
+
+      - name: Run heartbeat test
+        run: |
+          python actions-hub/scripts/smoke-tests/test_heartbeat.py \
+            --base-url "${{ inputs.target_url }}"
+
+  validate-plugins:
+    if: ${{ inputs.run_test == 'All' ||  inputs.run_test == 'validate-plugins'}}
+    name: Validate eox plugins
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout actions-hub repo
+        uses: actions/checkout@v4
+        with:
+          repository: nelc/actions-hub
+          ref: and/add-basic-smoke-tests
+          path: actions-hub
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+
+      - name: Install dependencies
+        run: |
+          pip install pytest-playwright
+          playwright install
+
+      - name: Validate eox plugins
+        run: |
+          python actions-hub/scripts/smoke-tests/test_eox_plugins.py \
+            --base-url "${{ inputs.target_url }}" \
+            --plugins ${{ inputs.target_plugins }}
+
+  login-user:
+    if: ${{ inputs.run_test == 'All' ||  inputs.run_test == 'login-user'}}
+    name: Login User Test
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout actions-hub repo
+        uses: actions/checkout@v4
+        with:
+          repository: nelc/actions-hub
+          ref: and/add-basic-smoke-tests
+          path: actions-hub
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+
+      - name: Install dependencies
+        run: |
+          pip install pytest-playwright
```

Comment:
Why do you select playwright instead of selenium??

Reply:
And why do you select these tools that use Chromium instead of other tools like K6 or Locust (load testing tools)? To test the browser behavior??